File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height) - 1
        water = 0

        while left < right:
            h = min(height[left], height[right])
            w = right - left
            area = h*w
            water = max(water, area)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1
        
        return water

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0; 
        int right = height.size()-1;

        int rain = 0;

        while(left < right){
            int h = min(height[left], height[right]);
            int w = right - left;

            int area = h*w;
            rain = max(area, rain);

            if(height[left] <= height[right]){
                left++;
            }else{
                right--;
            }
        }

        return rain;
    }
};